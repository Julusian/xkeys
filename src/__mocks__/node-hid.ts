import { EventEmitter } from 'events'

export interface Device {
	vendorId: number
	productId: number
	path?: string
	serialNumber?: string
	manufacturer?: string
	product?: string
	release: number
	interface: number
	usagePage?: number
	usage?: number
}

// export class HID extends EventEmitter {
export class HID extends EventEmitter {
	constructor (_path: string) {
		super()
	}
	// constructor(vid: number, pid: number);
	close(): void {
		// void
	}
	pause(): void {
		// void
	}
	read(_callback: (err: any, data: number[]) => void): void {
		// void
	}
	readSync(): number[] {
		return []
	}
	readTimeout(_timeOut: number): number[] {
		return []
	}
	sendFeatureReport(_data: number[]): number {
		return 0
	}
	getFeatureReport(_reportIdd: number, _reportLength: number): number[] {
		return []
	}
	resume(): void {
		// void
	}
	write(message: number[]): number {
		this.mockWriteHandler?.(this, message)
		return 0
	}
	setNonBlocking(_noBlock: boolean): void {
		// void
	}
}
export function devices(): Device[] {
	return []
}
export function setDriverType(_type: 'hidraw' | 'libusb'): void {
	// void
}
