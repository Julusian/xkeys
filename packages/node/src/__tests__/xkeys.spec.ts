import * as HID from 'node-hid'
import * as HIDMock from '../__mocks__/node-hid'
import { setupXkeysPanel, XKeys } from '../'
import { getSentData, handleXkeysMessages, resetSentData } from './lib'

describe('Unit tests', () => {
	test('calculateDelta', () => {
		expect(XKeys.calculateDelta(100, 100)).toBe(0)
		expect(XKeys.calculateDelta(110, 100)).toBe(10)
		expect(XKeys.calculateDelta(90, 100)).toBe(-10)
		expect(XKeys.calculateDelta(0, 255)).toBe(1)
		expect(XKeys.calculateDelta(5, 250)).toBe(11)
		expect(XKeys.calculateDelta(255, 0)).toBe(-1)
		expect(XKeys.calculateDelta(250, 5)).toBe(-11)
	})
})
