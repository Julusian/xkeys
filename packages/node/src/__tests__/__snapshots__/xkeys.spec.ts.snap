// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests XKeys methods 1`] = `
Object {
  "colCount": 4,
  "emitsTimestamp": true,
  "firmwareVersion": 1,
  "hasDMX": false,
  "hasGPIO": false,
  "hasJog": 0,
  "hasJoystick": 0,
  "hasLCD": false,
  "hasPS": true,
  "hasRotary": 0,
  "hasSerialData": false,
  "hasShuttle": 0,
  "hasTbar": 0,
  "hasTrackball": 0,
  "interface": 0,
  "layout": Array [
    Object {
      "endCol": 4,
      "endRow": 6,
      "index": 0,
      "name": "Keys",
      "startCol": 1,
      "startRow": 1,
    },
  ],
  "name": "XK-24",
  "productId": 1029,
  "rowCount": 6,
  "unitId": 0,
  "vendorId": 1523,
}
`;

exports[`Unit tests XKeys methods 2`] = `Array []`;

exports[`Unit tests XKeys methods 3`] = `
Array [
  "00b305010000000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 4`] = `
Array [
  "00b305000000000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 5`] = `
Array [
  "00b305020000000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 6`] = `
Array [
  "00b504010100000000000000000000000000000000000000000000000000000000000000",
  "00b524010100000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 7`] = `
Array [
  "00b504010100000000000000000000000000000000000000000000000000000000000000",
  "00b524010100000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 8`] = `
Array [
  "00b504010100000000000000000000000000000000000000000000000000000000000000",
  "00b524010100000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 9`] = `
Array [
  "00b504010100000000000000000000000000000000000000000000000000000000000000",
  "00b524010100000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 10`] = `
Array [
  "00b504010100000000000000000000000000000000000000000000000000000000000000",
  "00b524000100000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 11`] = `
Array [
  "00b504000100000000000000000000000000000000000000000000000000000000000000",
  "00b524000100000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 12`] = `
Array [
  "00b504000100000000000000000000000000000000000000000000000000000000000000",
  "00b524000100000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 13`] = `
Array [
  "00b504000100000000000000000000000000000000000000000000000000000000000000",
  "00b524000100000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 14`] = `
Array [
  "00b504020100000000000000000000000000000000000000000000000000000000000000",
  "00b524020100000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 15`] = `
Array [
  "00b600ff0000000000000000000000000000000000000000000000000000000000000000",
  "00b601550000000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 16`] = `
Array [
  "00b600ff0000000000000000000000000000000000000000000000000000000000000000",
  "00b601550000000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 17`] = `
Array [
  "00b600ff0000000000000000000000000000000000000000000000000000000000000000",
  "00b601550000000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 18`] = `
Array [
  "00b600ff0000000000000000000000000000000000000000000000000000000000000000",
  "00b6012d0000000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 19`] = `
Array [
  "00b600ff0000000000000000000000000000000000000000000000000000000000000000",
  "00b601000000000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 20`] = `
Array [
  "00b600000000000000000000000000000000000000000000000000000000000000000000",
  "00b601000000000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 21`] = `
Array [
  "00b600000000000000000000000000000000000000000000000000000000000000000000",
  "00b601000000000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 22`] = `
Array [
  "00b600000000000000000000000000000000000000000000000000000000000000000000",
  "00b601000000000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 23`] = `
Array [
  "00b800000000000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 24`] = `
Array [
  "00bb64640000000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 25`] = `
Array [
  "00bb00ff0000000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 26`] = `
Array [
  "00c701000000000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 27`] = `
Array [
  "00b47f000000000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 28`] = `
Array [
  "00bd2a000000000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 29`] = `
Array [
  "00ee00000000000000000000000000000000000000000000000000000000000000000000",
]
`;

exports[`Unit tests XKeys methods 30`] = `
Array [
  "000102030400000000000000000000000000000000000000000000000000000000000000",
]
`;
